app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,60 @@
+class PasswordResetsController < ApplicationController
+  before_action :get_user, only: %i(edit update)
+  before_action :valid_user, only: %i(edit update)
+  before_action :check_expiration, only: %i(edit update)
```

Comment:
For all of those filters, as the actions are same, you can combine line 4, 3 and 2 into one